python/ray/_private/custom_types.py

```diff
@@ -72,9 +71,19 @@
     "DRIVER_TASK",
 ]
 TypeTaskType = Literal[tuple(TASK_TYPE)]
-TypeReferenceType = Literal[
-    tuple(reference_type.value for reference_type in ReferenceType)
+# TODO(kevin85421): `class ReferenceType(Enum)` is defined in
```

Comment:
To make this PR easier to review, I will open a follow up PR to address this instead of including in this PR.

https://github.com/ray-project/ray/blob/766a525a4d2ab234af0c5525724919ba081cc9bd/python/ray/dashboard/memory_utils.py#L47